package.json

```diff
@@ -20,56 +20,56 @@
     "dist"
   ],
   "engines": {
-    "node": ">=16"
+    "node": ">=20"
   },
   "resolutions": {
     "wrap-ansi": "7.0.0"
   },
   "dependencies": {
-    "@aws-sdk/client-cloudfront": "^3.224.0",
-    "@aws-sdk/client-lambda": "^3.224.0",
-    "@aws-sdk/client-secrets-manager": "^3.468.0",
-    "punycode": "^2.1.1"
+    "@aws-sdk/client-cloudfront": "^3.515.0",
+    "@aws-sdk/client-lambda": "^3.515.0",
+    "@aws-sdk/client-secrets-manager": "^3.515.0"
   },
   "devDependencies": {
-    "@commitlint/config-conventional": "^17.0.3",
-    "@rollup/plugin-commonjs": "^24.0.0",
-    "@rollup/plugin-json": "^4.1.0",
-    "@rollup/plugin-node-resolve": "^15.0.1",
-    "@rollup/plugin-replace": "^5.0.1",
-    "@rollup/plugin-typescript": "^9.0.2",
-    "@types/aws-lambda": "^8.10.107",
-    "@types/jest": "^29.2.0",
-    "@types/node": "^18.11.9",
-    "@types/punycode": "^2.1.0",
-    "@typescript-eslint/eslint-plugin": "^5.44.0",
-    "@typescript-eslint/parser": "^5.44.0",
+    "@commitlint/cli": "^18.6.1",
+    "@fingerprintjs/commit-lint-dx-team": "^0.0.2",
+    "@fingerprintjs/conventional-changelog-dx-team": "^0.1.0",
+    "@fingerprintjs/eslint-config-dx-team": "^0.1.0",
+    "@fingerprintjs/prettier-config-dx-team": "^0.1.0",
+    "@fingerprintjs/tsconfig-dx-team": "^0.0.2",
+    "@rollup/plugin-commonjs": "^25.0.7",
+    "@rollup/plugin-json": "^6.1.0",
+    "@rollup/plugin-node-resolve": "^15.2.3",
+    "@rollup/plugin-replace": "^5.0.5",
+    "@rollup/plugin-typescript": "^11.1.6",
+    "@types/aws-lambda": "^8.10.133",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^20.11.19",
     "adm-zip": "^0.5.10",
-    "aws-sdk-client-mock": "^3.0.0",
-    "aws-sdk-client-mock-jest": "^3.0.0",
-    "commitizen": "^4.2.4",
-    "conventional-changelog-conventionalcommits": "5.0.0",
+    "aws-sdk-client-mock": "^3.0.1",
+    "aws-sdk-client-mock-jest": "^3.0.1",
+    "commitizen": "^4.3.0",
+    "conventional-changelog-conventionalcommits": "7.0.2",
     "cz-conventional-changelog": "^3.3.0",
-    "dotenv": "^16.0.3",
-    "eslint": "^8.28.0",
-    "eslint-config-prettier": "^8.5.0",
-    "eslint-plugin-prettier": "^4.2.1",
-    "husky": "^8.0.1",
-    "jest": "^29.2.2",
-    "jest-environment-jsdom": "^29.2.2",
-    "lint-staged": "^13.0.3",
-    "prettier": "^2.8.0",
-    "rollup": "^2.70.1",
-    "rollup-plugin-dts": "^5.0.0",
-    "rollup-plugin-license": "^2.8.1",
-    "ts-jest": "^29.0.3",
-    "tslib": "^2.4.1",
-    "typescript": "^4.9.3",
-    "whatwg-fetch": "^3.6.2"
+    "dotenv": "^16.4.4",
+    "husky": "^8.0.3",
+    "jest": "^29.7.0",
+    "jest-environment-jsdom": "^29.7.0",
+    "lint-staged": "^15.2.2",
+    "prettier": "^3.2.5",
+    "rimraf": "^5.0.5",
+    "rollup": "^4.11.0",
```

Comment:
The Rollup version is changed 2 MAJOR versions here, from v2 to v4. This is pretty aggressive. Why are we making this change, and how are we sure the whole build system has no problems?

This also applies to other packages too, but rollup is the first one I wanted to ask.

Reply:
The easiest way to find out is to compare artifacts generated with an old and with a new version.
I did it for the react SDK already and I was satisfied by the result.

Reply:
Yes, but one also needs to read the changelog at least to make sure no unexpected changes are in the new version. My question is about sharing what has been done in terms of researching its effect.

Reply:
> The most important change for v4 is that minimal Node v18.

This is enough for me. Thanks!